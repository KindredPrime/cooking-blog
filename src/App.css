body {
  text-align: center;
  margin: 0;
}

main {
  margin: 0 10% 0 10%;
}