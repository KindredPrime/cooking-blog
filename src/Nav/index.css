.Nav {
  background-color: lightgray;
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  display: flex;
  justify-content: flex-end;
  padding: 20px 5% 20px 5%;
}

.Nav a {
  margin-left: 20px;
}

.Nav a.active {
  font-weight: bold;
}