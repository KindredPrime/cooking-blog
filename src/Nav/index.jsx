import { NavLink } from 'react-router-dom';
import './index.css';

function Nav() {
  return (
    <nav className="Nav">
      <NavLink exact to="/">Home</NavLink>
      <NavLink to="/account">Account</NavLink>
    </nav>
  );
}

export default Nav;