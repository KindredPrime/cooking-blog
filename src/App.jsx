import { Route } from 'react-router-dom';
import Nav from './Nav/index';
import Home from './Home/index';
import Account from './Account/index';
import CookingContext from './CookingContext';
import './App.css';

function App() {
  const contextValue = {
    username: 'User1'
  };

  return (
    <CookingContext.Provider value={contextValue}>
      <div className="App">
        <Nav />

        <Route exact path="/" component={Home} />
        <Route
          path="/account"
          // The username and email are set to a default for the static client
          render={() => <Account username="User1" email="user1@gmail.com" />}
        />
      </div>
    </CookingContext.Provider>
  );
}

export default App;