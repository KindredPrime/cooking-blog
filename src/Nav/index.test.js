import ReactDOM from 'react-dom';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Nav from './index';
import App from '../App';

describe('Nav Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('loads the homepage when clicked', () => {
    render(
      <MemoryRouter initialEntries={['/account']}>
        <App />
      </MemoryRouter>
    );

    userEvent.click(screen.getByText('Home'));
    expect(document.body).toMatchSnapshot();
  });

  it('loads the account page when clicked', () => {
    render(
      <MemoryRouter initialEntries={['/']}>
        <App />
      </MemoryRouter>
    );

    userEvent.click(screen.getByText('Account'));
    expect(document.body).toMatchSnapshot();
  });
});