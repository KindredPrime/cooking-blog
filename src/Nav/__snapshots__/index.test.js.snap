// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component loads the account page when clicked 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <nav
        class="Nav"
      >
        <a
          href="/"
        >
          Home
        </a>
        <a
          aria-current="page"
          class="active"
          href="/users/1"
        >
          Account
        </a>
      </nav>
      <main
        class="UserPage"
      >
        <header>
          <h1>
            User1
          </h1>
        </header>
        <p>
          Email: 
          user1@gmail.com
        </p>
        <section
          class="PostsList"
        >
          <header>
            <h2>
              Posts
            </h2>
          </header>
          <ul>
            <li
              class="PostsList__blog-post"
            >
              <a
                class="PostsList__title"
                href="/blog-posts/4"
              >
                Cooking Basics for Beginners
              </a>
              <p
                class="timestamp"
              >
                Last edited: 
                Jan 05 2021 08:28:28
              </p>
            </li>
            <li
              class="PostsList__blog-post"
            >
              <a
                class="PostsList__title"
                href="/blog-posts/2"
              >
                How to Save Money When Cooking
              </a>
              <p
                class="timestamp"
              >
                Last edited: 
                Dec 28 2020 18:24:43
              </p>
            </li>
          </ul>
        </section>
        <section
          class="UserCommentsList"
        >
          <header>
            <h2>
              Comments
            </h2>
          </header>
          <ul>
            <li
              class="Comment"
            >
              <a
                class="Comment__post-title"
                href="/blog-posts/1"
              >
                Lessons I Learned from Cooking
              </a>
              <p
                class="Comment__content"
              >
                These are all great lessons!
              </p>
              <p
                class="timestamp"
              >
                Last edited: 
                Dec 19 2020 07:37:16
              </p>
            </li>
            <li
              class="Comment"
            >
              <a
                class="Comment__post-title"
                href="/blog-posts/5"
              >
                Winter Seasoning Guide
              </a>
              <p
                class="Comment__content"
              >
                I can't wait to try these seasoning combinations in my next meal!
              </p>
              <p
                class="timestamp"
              >
                Last edited: 
                Dec 17 2020 08:37:52
              </p>
            </li>
          </ul>
        </section>
      </main>
    </div>
  </div>
</body>
`;

exports[`Nav Component loads the homepage when clicked 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <nav
        class="Nav"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
        <a
          href="/users/1"
        >
          Account
        </a>
      </nav>
      <main
        class="Home"
      >
        <header>
          <h1>
            Welcome to the Cooking Blog!
          </h1>
        </header>
        <p>
          Ever wanted to learn more about cooking, or share your experiences with others? Here you can read all kinds of posts from other users: things they've tried, things they want to try, recipes, cooking tips, etc. And if you create an account you can make posts of your own, and leave comments on posts.
        </p>
        <section
          class="PostsList"
        >
          <header>
            <h2>
              Posts
            </h2>
          </header>
          <ul>
            <li
              class="PostsList__blog-post"
            >
              <a
                class="PostsList__title"
                href="/blog-posts/8"
              >
                Title8
              </a>
              <p
                class="timestamp"
              >
                Last edited: 
                Jan 06 2021 12:22:39
              </p>
            </li>
            <li
              class="PostsList__blog-post"
            >
              <a
                class="PostsList__title"
                href="/blog-posts/6"
              >
                Title6
              </a>
              <p
                class="timestamp"
              >
                Last edited: 
                Jan 05 2021 14:22:39
              </p>
            </li>
            <li
              class="PostsList__blog-post"
            >
              <a
                class="PostsList__title"
                href="/blog-posts/4"
              >
                Cooking Basics for Beginners
              </a>
              <p
                class="timestamp"
              >
                Last edited: 
                Jan 05 2021 08:28:28
              </p>
            </li>
            <li
              class="PostsList__blog-post"
            >
              <a
                class="PostsList__title"
                href="/blog-posts/11"
              >
                Title11
              </a>
              <p
                class="timestamp"
              >
                Last edited: 
                Jan 04 2021 18:22:39
              </p>
            </li>
            <li
              class="PostsList__blog-post"
            >
              <a
                class="PostsList__title"
                href="/blog-posts/7"
              >
                Title7
              </a>
              <p
                class="timestamp"
              >
                Last edited: 
                Jan 02 2021 16:22:39
              </p>
            </li>
            <li
              class="PostsList__blog-post"
            >
              <a
                class="PostsList__title"
                href="/blog-posts/2"
              >
                How to Save Money When Cooking
              </a>
              <p
                class="timestamp"
              >
                Last edited: 
                Dec 28 2020 18:24:43
              </p>
            </li>
            <li
              class="PostsList__blog-post"
            >
              <a
                class="PostsList__title"
                href="/blog-posts/1"
              >
                Lessons I Learned from Cooking
              </a>
              <p
                class="timestamp"
              >
                Last edited: 
                Dec 17 2020 07:37:16
              </p>
            </li>
            <li
              class="PostsList__blog-post"
            >
              <a
                class="PostsList__title"
                href="/blog-posts/10"
              >
                Title10
              </a>
              <p
                class="timestamp"
              >
                Last edited: 
                Dec 15 2020 19:22:39
              </p>
            </li>
            <li
              class="PostsList__blog-post"
            >
              <a
                class="PostsList__title"
                href="/blog-posts/5"
              >
                Winter Seasoning Guide
              </a>
              <p
                class="timestamp"
              >
                Last edited: 
                Dec 15 2020 11:56:48
              </p>
            </li>
            <li
              class="PostsList__blog-post"
            >
              <a
                class="PostsList__title"
                href="/blog-posts/9"
              >
                Title9
              </a>
              <p
                class="timestamp"
              >
                Last edited: 
                Dec 15 2020 11:22:39
              </p>
            </li>
          </ul>
          <div
            class="TransitionButtons"
          >
            <button
              class="TransitionButtons__prev"
              disabled=""
              type="button"
            >
              Previous
            </button>
            <button
              class="TransitionButtons__next"
              type="button"
            >
              Next
            </button>
          </div>
        </section>
      </main>
    </div>
  </div>
</body>
`;
